core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -149,6 +151,24 @@ static <E> List<E> list() {
     assertThat(concatWithNull(null, "b"), is("b"));
   }
 
+  @Test void testConcatMulti() {
+    assertThat(concatMulti("a b", "cd", "e"), is("a bcde"));
+    // The code generator will ensure that nulls are never passed in. If we
+    // pass in null, it is treated like the string "null", as the following
+    // tests show. Not the desired behavior for SQL.
+    assertThat(concatMulti((String) null), is("null"));
+    assertThat(concatMulti((String) null, null), is("nullnull"));
+    assertThat(concatMulti("a", null, "b"), is("anullb"));
+  }
+
+  @Test void testConcatMultiWithNull() {
+    assertThat(concatMultiWithNull("a b", "cd", "e"), is("a bcde"));
+    // Null value could be passed in which is treated as empty string.
+    assertThat(concatMultiWithNull((String) null), is(""));
```

Comment:
Just to clarify, which of these tests mimics the behavior that BigQuery follows? My understanding is that BigQuery does the following:

`CONCAT(null, null)` returns `NULL` while
`CONCAT("", "")` returns `""`

Reply:
For BigQuery, you should see testConcatMulti(), and you are right about the behavior.(but the code generator will ensure nulls won't be passed in)
testConcatMultiWithNull() is a test for CONCAT function that enabled in Postgresql and MSSQL.